src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/CanonicalDefinitionEETypeNode.cs

```diff
@@ -17,16 +18,25 @@ public CanonicalDefinitionEETypeNode(NodeFactory factory, TypeDesc type) : base(
             Debug.Assert(type.IsCanonicalDefinitionType(CanonicalFormKind.Any));
         }
 
-        public override bool ShouldSkipEmittingObjectNode(NodeFactory factory) => false;
```

Comment:
Why? This likely breaks multimodule builds. In multimodule builds, we expect __Canon and __UniversalCanon to be generated into each module that needs it and be COMDAT folded at link time (`IsShareable => true` below).